### DISCUSSION TITLE
Building a docsite outside the project folder and committing to branch

### DISCUSSION THREAD

**Discussion by exoRift** [community/community#26934] [Actions]
<pre><code>name: Update Docsite

on:
  create:
    tags:
      - v*

jobs:
  build:
    name: Build Docsite

    strategy:
      matrix:
        os: [ubuntu-latest]

    runs-on: ${{ matrix.os }}

    steps:
    - uses: actions/checkout@v2
    - name: Cache Dependencies
      uses: actions/cache@v1.1.2
      env:
        cache-name: cache-node-modules
      with:
        path: ~/.npm
        key: ${{ matrix.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
        restore-keys: |
          ${{ runner.os }}-build-${{ env.cache-name }}-
          ${{ runner.os }}-build-
          ${{ runner.os }}-
    - name: Install Dependencies
      run: npm i
    - name: Define package data
      id: define_pkg
      run: |
        content=`cat **/package.json`
        echo "::set-output name=package::$content"
    - name: Initiate Git
      run: |
        git config --global user.name "Doc updater"
        cd ../cyclone-engine-pages
        git init
        git remote add origin https://github.com/exoRift/cyclone-engine.git
        git fetch origin gh-pages
    - name: Build Resources
      run: npm run build
    - name: Commit Docsite
      run: |
        cd ../cyclone-engine-pages
        git commit -m "Update docsite for version '${{ fromJson(steps.define_pkg.outputs.package).version }}'"
</code></pre>
<p>So, I have this workflow that’s a work in progress. <code>npm run build</code> generates a docsite in the directory <code>../cyclone-engine-pages</code> which I initialized and fetched before as a branch of my repo.</p>
<p>So I have 3 questions with this code<br>
1: Can I manipulate the filesystem outside the project folder directory?<br>
2: Do I have to reset my CD to the project folder for step “Build Resources”?</p>
<p>3: Does this workflow work?</p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/exorift">@exorift</a> ,</p>
<p>Thank you for being here! To answer your queries:</p>
<ol>
<li>
<p>yes, you can manipulate the filesystem outside the project folder directory. For github hosted runner, you need to manually create the folder before you cd into.</p>
<ul>
<li>name: Initiate Git<br>
run: |<br>
mkdir …/cyclone-engine-pages cd …/cyclone-engine-pages</li>
</ul>
</li>
<li>
<p>The default working directory is ‘GITHUB_WORKSPACE’, on ubuntu hosted runner, which typically stands for ‘/home/runner/work/repo/repo’, you can check with command ‘pwd’.</p>
</li>
</ol>
<p>For the step ‘Build Resources’, the ‘npm run build’ command is executed under the GITHUB_WORKSPACE directory NOT ‘cyclone-engine-pages’.</p>
<ol start="3">
<li>For ‘Commit Docsite’ step, there is no change to commit since npm built in the GITHUB_WORKSPACE directory.</li>
</ol>
<p>Besides, github hosted runner will be released after workflow completed, if you only commit but not push back the change, no change will be saved after workflow.  </p>
<p>You can use action ‘ad-m/github-push-action’ to push the change, please check the <a href="https://github.com/ad-m/github-push-action#usage" rel="nofollow noopener">usage</a> of the action.</p>
<p>Hence, your workflow should be not working.</p>
<p>Besides, using the working-directory keyword, you can specify the working directory of where to run the command. Please check the <a href="https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstepsrun" rel="nofollow noopener">doc</a> for more details.</p>
<p>Thanks.</p>

**Comment by exoRift** [comment]
<p>Thank you for the very clear answers! I forgot about <code>mkdir</code>.</p>
<p>Just 2 things to note,</p>
<p>1: <code>npm run build</code> exports its contents to <code>../cyclone-engine-pages</code>, so I’m good.<br>
2: Yeah, I haven’t implemented the push code yet, but I will. Thank you for your help!</p>
